File: src/main/java/com/example/demo/service/ManufacturerService.java
package com.example.demo.service;

import com.example.demo.dto.DictionaryData;
import com.example.demo.dto.DictionaryDataInput;

import java.util.List;

public interface ManufacturerService {
    DictionaryData addManufacturer(DictionaryDataInput dictionaryDataInput);
    List<DictionaryData> findAllManufacturers();
    DictionaryData updateManufacturer(DictionaryData dictionaryData);
    DictionaryData findManufacturerById(Long id);
    void deleteManufacturer(Long id);
}

File: WEB/gun-mgmt-app/src/app/services/manufacturer.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { DictionaryData } from '../models/DictionaryData';
import { DictionaryDataInput } from '../models/DictionaryDataInput';

@Injectable({
  providedIn: 'root'
})
export class ManufacturerService {
  private apiEndpointUrl = `${environment.apiUrl}/manufacturers`;

  constructor(private http: HttpClient) { }

  public getAllManufacturers() :Observable<DictionaryData[]> {
    return this.http.get<DictionaryData[]>(`${this.apiEndpointUrl}`);
  }

  public getManufacturerById(id: number) :Observable<DictionaryData> {
    return this.http.get<DictionaryData>(`${this.apiEndpointUrl}/${id}`);
  }

  public addManufacturer(manufacturerInput: DictionaryDataInput) :Observable<DictionaryData> {
    return this.http.post<DictionaryData>(`${this.apiEndpointUrl}/add`, manufacturerInput);
  }

  public updateManufacturer(manufacturer: DictionaryData) :Observable<DictionaryData> {
    return this.http.put<DictionaryData>(`${this.apiEndpointUrl}/update`, manufacturer);
  }

  public deleteManufacturer(id: number) :Observable<void> {
    return this.http.delete<void>(`${this.apiEndpointUrl}/delete/${id}`);
  }
}
